shell/browser/native_window_mac.mm

```diff
@@ -1351,8 +1351,21 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
 void NativeWindowMac::SetOverlayIcon(const gfx::Image& overlay,
                                      const std::string& description) {}
 
-void NativeWindowMac::SetVisibleOnAllWorkspaces(bool visible) {
+void NativeWindowMac::SetVisibleOnAllWorkspaces(bool visible,
+                                                bool visibleOnFullScreen) {
+  // In order for NSWindows to be visible on fullscreen we need to functionally
+  // mimic app.dock.hide() since Apple changed the underlying functionality of
+  // NSWindows starting with 10.14 to disallow NSWindows from floating on top of
+  // fullscreen apps.
+  if (visibleOnFullScreen) {
```

Comment:
Don't we need to do the inverse of this when `visibleOnFullScreen=false`? i.e. we need an `else` block